include/alpaka/extent/Traits.hpp

```diff
@@ -22,121 +23,183 @@ namespace alpaka
         //!
         //! If not specialized explicitly it returns 1.
         template<typename TIdxIntegralConst, typename TExtent, typename TSfinae = void>
-        struct GetExtent
-        {
-            ALPAKA_NO_HOST_ACC_WARNING
-            ALPAKA_FN_HOST_ACC static auto getExtent(TExtent const&) -> Idx<TExtent>
-            {
-                return static_cast<Idx<TExtent>>(1);
-            }
-        };
-
-        //! The extent set trait.
-        template<typename TIdxIntegralConst, typename TExtent, typename TExtentVal, typename TSfinae = void>
-        struct SetExtent;
+        struct [[deprecated("Specialize GetExtents instead")]] GetExtent{
+            ALPAKA_NO_HOST_ACC_WARNING ALPAKA_FN_HOST_ACC static auto getExtent(TExtent const&)
+                ->Idx<TExtent>{return static_cast<Idx<TExtent>>(1);
     } // namespace trait
+}; // namespace alpaka
 
-    //! \return The extent in the given dimension.
-    ALPAKA_NO_HOST_ACC_WARNING
-    template<std::size_t Tidx, typename TExtent>
-    ALPAKA_FN_HOST_ACC auto getExtent(TExtent const& extent = TExtent()) -> Idx<TExtent>
-    {
-        return trait::GetExtent<DimInt<Tidx>, TExtent>::getExtent(extent);
-    }
-    //! \return The width.
-    ALPAKA_NO_HOST_ACC_WARNING
-    template<typename TExtent>
-    ALPAKA_FN_HOST_ACC auto getWidth(TExtent const& extent = TExtent()) -> Idx<TExtent>
-    {
-        return getExtent<Dim<TExtent>::value - 1u>(extent);
-    }
-    //! \return The height.
-    ALPAKA_NO_HOST_ACC_WARNING
-    template<typename TExtent>
-    ALPAKA_FN_HOST_ACC auto getHeight(TExtent const& extent = TExtent()) -> Idx<TExtent>
-    {
-        return getExtent<Dim<TExtent>::value - 2u>(extent);
-    }
-    //! \return The depth.
-    ALPAKA_NO_HOST_ACC_WARNING
-    template<typename TExtent>
-    ALPAKA_FN_HOST_ACC auto getDepth(TExtent const& extent = TExtent()) -> Idx<TExtent>
-    {
-        return getExtent<Dim<TExtent>::value - 3u>(extent);
-    }
+//! The GetExtents trait for getting the extents of an object as an alpaka::Vec.
+template<typename TExtent, typename TSfinae = void>
+struct GetExtents;
+
+//! The extent set trait.
+template<typename TIdxIntegralConst, typename TExtent, typename TExtentVal, typename TSfinae = void>
+struct SetExtent;
+} // namespace trait
+
+//! \return The extent in the given dimension.
+ALPAKA_NO_HOST_ACC_WARNING
+template<std::size_t Tidx, typename TExtent>
+[[deprecated("use getExtents(extent)[Tidx] instead")]] ALPAKA_FN_HOST_ACC auto getExtent(
+    TExtent const& extent = TExtent()) -> Idx<TExtent>
+{
+#if BOOST_COMP_CLANG || BOOST_COMP_GNUC
+#    pragma GCC diagnostic push
+#    pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#endif
```

Comment:
What is this? We are using `[[deprecated]]` but are turning off the related warning?

Reply:
@j-stephan I don't know what you mean with your thumbs up. Keep deprecated functions or remove them?

Reply:
Oh, sorry. This was just in reference to your explanation. Keep the deprecated version for now, we don't want to immediately break things on the user side.